development/build/transforms/utils.js

```diff
@@ -0,0 +1,65 @@
+const { ESLint } = require('eslint');
+const eslintrc = require('../../../.eslintrc.js');
+
+// This is a singleton
+let eslintInstance;
+
+// Only initialize the ESLint instance if necessary
+// This also makes it easier to test this module
+const initializeESLint = () => {
+  if (!eslintInstance) {
+    eslintInstance = new ESLint({ baseConfig: eslintrc, useEslintrc: false });
+  }
+};
+
+// Four spaces
+const TAB = '    ';
+
+module.exports = {
+  lintTransformedFile,
+};
+
+/**
+ * Lints a transformed file by invoking ESLint programmatically on the string
+ * file contents. The path to the file must be specified so that the repository
+ * ESLint config can be applied properly.
+ *
+ * An error is thrown if linting produced any errors, or if the file is ignored
+ * by ESLint. Files linted by this function should never be ignored.
+ *
+ * @param {string} content - The file content.
+ * @param {string} filePath - The path to the file.
+ * @returns {Promise<void>} Returns `undefined` or throws an error if linting produced
+ * any errors, or if the linted file is ignored.
+ */
+async function lintTransformedFile(content, filePath) {
+  initializeESLint();
+
+  const lintResult = (
+    await eslintInstance.lintText(content, { filePath, warnIgnored: false })
+  )[0];
+
+  // This indicates that the file is ignored, which should never be the case for
+  // a transformed file.
+  if (lintResult === undefined) {
+    throw new Error(
+      `MetaMask build: Transformed file "${filePath}" appears to be ignored by ESLint.`,
+    );
+  }
+
+  // This is the success case
+  if (lintResult.errorCount === 0) {
+    return;
+  }
+
+  // Errors are stored in the messages array, and their "severity" is 2
+  const errorsString = lintResult.messages
+    .filter(({ severity }) => severity === 2)
```

Comment:
should severity be checked via `>=` ?

Reply:
🙅 

[`ESLint.lintText`](https://eslint.org/docs/developer-guide/nodejs-api#-eslintlinttextcode-options) returns an array of [`LintResult` objects](https://eslint.org/docs/developer-guide/nodejs-api#-lintresult-type), and the `messages` property of such objects is an array of objects with a `severity` property that is [either the number `1` or `2`](https://eslint.org/docs/developer-guide/nodejs-api#-lintmessage-type).